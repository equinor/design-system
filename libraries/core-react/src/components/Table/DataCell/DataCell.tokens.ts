import { tokens } from '@equinor/eds-tokens'
import * as R from 'ramda'
import type { ComponentToken } from '@equinor/eds-tokens'
import { Variants } from '../Table.types'

const {
  typography: {
    table: {
      cell_text: cellTypography,
      cell_numeric_monospaced: cellNumericTypography,
    },
    _modes: { compact: compactTypography },
  },
  colors: {
    text: {
      static_icons__default: { rgba: typographyColor },
    },
    ui: {
      background__medium: { rgba: borderColor },
    },
    interactive: {
      table__cell__fill_resting: { rgba: backgroundColor },
      table__cell__fill_hover: { rgba: hoverBackgroundColor },
      table__cell__fill_activated: { rgba: activeBackgroundColor },
      danger__highlight: { rgba: errorColor },
      primary__resting: { rgba: primaryRestingColor },
      disabled__text: { rgba: disabledTextColor },
      disabled__border: { rgba: disabledBorderColor },
      focus: { rgba: focusColor },
    },
  },
  spacings: {
    comfortable: { medium },
  },
} = tokens

type Validation = {
  error: Partial<ComponentToken>
}

type VariantsType = {
  numeric: Partial<ComponentToken>
}

export type TableCellToken = ComponentToken & {
  validation: Validation
  variants: VariantsType
}

export const tableCell: TableCellToken = {
  height: '48px',
  background: backgroundColor,
  border: {
    type: 'bordergroup',
    bottom: {
      color: borderColor,
      width: '1px',
      style: 'solid',
    },
  },
  spacings: {
    top: '0',
    bottom: '0',
    left: medium,
    right: medium,
  },
  typography: {
    ...cellTypography,
    color: typographyColor,
  },
  states: {
    active: {
      background: activeBackgroundColor,
      typography: {
        ...cellTypography,
        color: primaryRestingColor,
      },
      border: {
        bottom: {
          color: primaryRestingColor,
        },
      },
    },
    disabled: {
      typography: {
        ...cellTypography,
        color: disabledTextColor,
      },
      border: {
        type: 'bordergroup',
        bottom: {
          color: disabledBorderColor,
        },
      },
    },
    focus: {
      outline: {
        type: 'outline',
        color: focusColor,
        width: '1px',
        style: 'dashed',
      },
    },
    hover: {
      background: hoverBackgroundColor,
    },
  },
  modes: {
    compact: {
      height: '32px',
      typography: {
        ...compactTypography.table.cell_text,
        color: typographyColor,
      },
    },
  },
  validation: {
    error: {
      background: errorColor,
    },
  },
  variants: {
    numeric: {
      typography: { ...cellNumericTypography, color: typographyColor },
      modes: {
        compact: {
          typography: {
            ...compactTypography.table.cell_numeric_monospaced,
            color: typographyColor,
          },
        },
      },
    },
  },
}

export const applyVariant = (
  variant: Variants,
  token: TableCellToken,
): TableCellToken => {
  switch (variant) {
    case 'numeric':
      return R.mergeDeepRight(token, token.variants.numeric)

    default:
      return token
  }
}
