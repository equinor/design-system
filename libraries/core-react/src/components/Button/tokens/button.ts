import { tokens } from '@equinor/eds-tokens'
import { ButtonToken } from '../Button.types'

const {
  typography: {
    navigation: { button: buttonTypography },
  },
  colors: {
    interactive: {
      disabled__text: { rgba: disabledTextColor },
      disabled__border: { rgba: disabledBorderColor },
      disabled__fill: { rgba: disabledColor },
    },
  },
  spacings: {
    comfortable: { medium },
  },
  shape: {
    button: { minHeight: buttonHeight, borderRadius: buttonBorderRadius },
    _modes: {
      compact: {
        button: { minHeight: compactButtonHeight },
      },
    },
  },
  interactions: { focused: outline },
  clickbounds: {
    default__base: clicboundHeight,
    compact__standard: compactClickboundHeight,
  },
} = tokens

export const button: ButtonToken = {
  background: 'transparent',
  height: buttonHeight,
  typography: {
    ...buttonTypography,
    textAlign: 'center',
  },
  border: {
    type: 'border',
    width: '1px',
    color: 'transparent',
    radius: buttonBorderRadius,
    style: 'solid',
  },
  spacings: {
    left: medium,
    right: medium,
  },
  clickbound: {
    height: clicboundHeight,
    width: '100%',
    offset: {
      top: `${(parseInt(clicboundHeight) - parseInt(buttonHeight)) / 2 + 1}px`,
      left: '0',
    },
  },
  entities: {
    icon: {
      height: '24px',
      width: '24px',
    },
  },
  states: {
    hover: {
      border: {
        type: 'border',
        width: '1px',
        color: 'transparent',
        radius: buttonBorderRadius,
        style: 'solid',
      },
    },
    focus: {
      outline: {
        type: 'outline',
        offset: '2px',
        style: 'dashed',
        color: outline.color,
        width: outline.width,
      },
    },
    disabled: {
      background: disabledColor,
      border: {
        type: 'border',
        width: '1px',
        color: disabledBorderColor,
        style: 'solid',
      },
      typography: {
        ...buttonTypography,
        color: disabledTextColor,
        textAlign: 'center',
      },
    },
  },
  modes: {
    compact: {
      height: compactButtonHeight,
      clickbound: {
        height: compactClickboundHeight,
        width: '100%',
        offset: {
          top: `${
            (parseInt(compactClickboundHeight) -
              parseInt(compactButtonHeight)) /
              2 +
            1
          }px`,
          left: '0',
        },
      },
    },
  },
}
