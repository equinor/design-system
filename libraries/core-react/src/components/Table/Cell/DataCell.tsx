import * as React from 'react'
import { TdHTMLAttributes, forwardRef } from 'react'
import styled, { css } from 'styled-components'
import {
  typographyTemplate,
  bordersTemplate,
  spacingsTemplate,
} from '../../../utils'
import { token as tablecell, TableCellToken } from './DataCell.tokens'
import { Variants, Colors } from '../Table.types'
import { useEds } from '../../EdsProvider'

import { applyDensity } from './utils'

const applyMutations = (
  variant: Variants,
  token: TableCellToken,
): TableCellToken => {
  switch (variant) {
    case 'numeric':
      return {
        ...token,
        typography: {
          ...token.typography,
          ...token.variants.numeric.typography,
        },
      }

    default:
      return token
  }
}

type BaseProps = {
  token: TableCellToken
  color?: Colors
}

const Base = ({ token, color }: BaseProps) => {
  const { height, typography, spacings, border } = token

  const backgroundColor =
    color === 'error' ? token.validation.error?.background : ''

  const base = css`
    min-height: ${height};
    height: ${height};
    background: ${backgroundColor};
    ${spacingsTemplate(spacings)}
    ${typographyTemplate(typography)}
    ${bordersTemplate(border)}
  `
  return base
}

const StyledTableCell = styled.td`
  ${Base}
`

type CellProps = {
  /** Specifies which variant to use */
  variant?: Variants
  /** Specifies cell background color */
  color?: Colors
} & TdHTMLAttributes<HTMLTableDataCellElement>

export const TableDataCell = forwardRef<HTMLTableDataCellElement, CellProps>(
  function TableDataCell({ children, variant = 'text', ...rest }, ref) {
    const { density } = useEds()

    let token = applyMutations(variant, tablecell)
    token = applyDensity(density, token)

    return (
      <StyledTableCell token={token} {...rest} ref={ref}>
        {children}
      </StyledTableCell>
    )
  },
)
