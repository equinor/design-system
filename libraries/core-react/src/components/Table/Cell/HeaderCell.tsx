import * as React from 'react'
import { ThHTMLAttributes } from 'react'
import styled, { css } from 'styled-components'
import { typographyTemplate, spacingsTemplate, bordersTemplate } from '@utils'
import { token as tablehead, TableHeadToken } from './HeaderCell.tokens'
import { useTable } from '../Table.context'
import { applyDensity } from './utils'

type BaseProps = {
  token: TableHeadToken
} & Pick<React.AriaAttributes, 'aria-sort'>

const Base = (props: BaseProps) => {
  const { token } = props
  const { background, height, typography, spacings } = token
  const activeToken = token.states.active
  const ariaSort = props['aria-sort']
  let sortStylingHover = css({})
  let sortStylingActive = css({})

  if (ariaSort) {
    sortStylingHover = css`
      @media (hover: hover) and (pointer: fine) {
        &:hover {
          cursor: pointer;
          background: ${token.states.hover.background};
        }
      }
    `
  }

  if (ariaSort && ariaSort !== 'none') {
    sortStylingActive = css`
      ${activeToken.border.type === 'bordergroup'
        ? css`
            border-color: ${activeToken.border.bottom.color};
          `
        : ''};
      background: ${activeToken.background};
      color: ${activeToken.typography.color};
    `
  }

  const base = css`
    min-height: ${height};
    height: ${height};
    background: ${background};
    ${spacingsTemplate(spacings)}
    ${typographyTemplate(typography)}
    ${bordersTemplate(token.border)}
    ${sortStylingHover}
    ${sortStylingActive}
  `

  return base
}

const StyledTableCell = styled.th`
  ${Base}
`

const CellInner = styled.div`
  display: flex;
  align-items: center;
`

type CellProps = {
  sort?: React.AriaAttributes['aria-sort']
} & ThHTMLAttributes<HTMLTableHeaderCellElement>

export const TableHeaderCell = ({
  children,
  sort,
  ...rest
}: CellProps): JSX.Element => {
  const { density } = useTable()
  const token = applyDensity(density, tablehead)

  return (
    <StyledTableCell token={token} aria-sort={sort} {...rest}>
      <CellInner>{children}</CellInner>
    </StyledTableCell>
  )
}

// Cell.displayName = 'eds-table-cell'
