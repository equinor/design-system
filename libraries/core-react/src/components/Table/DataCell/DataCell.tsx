import { TdHTMLAttributes, forwardRef } from 'react'
import styled, { css, ThemeProvider } from 'styled-components'
import {
  typographyTemplate,
  bordersTemplate,
  spacingsTemplate,
} from '../../../utils'
import { tableCell, TableCellToken, applyVariant } from './DataCell.tokens'
import { Variants, Colors } from '../Table.types'
import { useEds } from '../../EdsProvider'
import { useToken } from '../../../hooks'

type BaseProps = {
  theme: TableCellToken
  color?: Colors
}

const StyledTableCell = styled.td(({ theme, color }: BaseProps) => {
  const { height, typography, spacings, border } = theme

  const backgroundColor =
    color === 'error' ? theme.validation.error?.background : ''

  const base = css`
    min-height: ${height};
    height: ${height};
    background: ${backgroundColor};
    ${spacingsTemplate(spacings)}
    ${typographyTemplate(typography)}
    ${bordersTemplate(border)}
  `
  return base
})

type CellProps = {
  /** Specifies which variant to use */
  variant?: Variants
  /** Specifies cell background color */
  color?: Colors
} & TdHTMLAttributes<HTMLTableDataCellElement>

export const TableDataCell = forwardRef<HTMLTableDataCellElement, CellProps>(
  function TableDataCell({ children, variant = 'text', ...rest }, ref) {
    const { density } = useEds()
    const token = useToken({ density }, applyVariant(variant, tableCell))()

    return (
      <ThemeProvider theme={token}>
        <StyledTableCell {...rest} ref={ref}>
          {children}
        </StyledTableCell>
      </ThemeProvider>
    )
  },
)
