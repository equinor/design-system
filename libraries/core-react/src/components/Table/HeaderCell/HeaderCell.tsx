import { ThHTMLAttributes, forwardRef } from 'react'
import styled, { css, ThemeProvider } from 'styled-components'
import {
  typographyTemplate,
  spacingsTemplate,
  bordersTemplate,
} from '../../../utils'
import { token as tablehead, TableHeadToken } from './HeaderCell.tokens'
import { useEds } from '../../EdsProvider'
import { useToken } from '../../../hooks'

type BaseProps = {
  theme: TableHeadToken
  sticky?: boolean
} & Pick<React.AriaAttributes, 'aria-sort'>

const StyledTableCell = styled.th((props: BaseProps) => {
  const { theme, sticky } = props
  const { background, height, typography, spacings } = theme
  const activeToken = theme.states.active
  const ariaSort = props['aria-sort']
  let sortStylingHover = css({})
  let sortStylingActive = css({})

  if (ariaSort) {
    sortStylingHover = css`
      @media (hover: hover) and (pointer: fine) {
        &:hover {
          cursor: pointer;
          background: ${theme.states.hover.background};
        }
      }
    `
  }

  if (ariaSort && ariaSort !== 'none') {
    sortStylingActive = css`
      ${activeToken.border.type === 'bordergroup'
        ? css`
            border-color: ${activeToken.border.bottom.color};
          `
        : ''};
      background: ${activeToken.background};
      color: ${activeToken.typography.color};
    `
  }

  return css`
    min-height: ${height};
    height: ${height};
    background: ${background};
    ${spacingsTemplate(spacings)}
    ${typographyTemplate(typography)}
  ${bordersTemplate(theme.border)}
  ${sortStylingHover}
  ${sortStylingActive}

  ${sticky
      ? css`
          position: sticky;
          top: 0;
        `
      : ''}
  `
})

const CellInner = styled.div`
  display: flex;
  align-items: center;
`

type CellProps = {
  sort?: React.AriaAttributes['aria-sort']
  sticky?: boolean
} & ThHTMLAttributes<HTMLTableHeaderCellElement>

export const TableHeaderCell = forwardRef<
  HTMLTableHeaderCellElement,
  CellProps
>(function TableHeaderCell({ children, sort, ...rest }, ref) {
  const { density } = useEds()
  const token = useToken({ density }, tablehead)()

  return (
    <ThemeProvider theme={token}>
      <StyledTableCell aria-sort={sort} {...rest} ref={ref}>
        <CellInner>{children}</CellInner>
      </StyledTableCell>
    </ThemeProvider>
  )
})
