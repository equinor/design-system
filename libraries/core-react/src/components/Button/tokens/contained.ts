import { tokens } from '@equinor/eds-tokens'
import { ButtonToken } from '../Button.types'
import { button } from './button'

import mergeDeepRight from 'ramda/src/mergeDeepRight'

const {
  colors: {
    text: {
      static_icons__primary_white: { rgba: primaryWhite },
    },
    interactive: {
      primary__resting: { rgba: primaryColor },
      primary__hover: { rgba: primaryHoverColor },
      secondary__resting: { rgba: secondaryColor },
      secondary__link_hover: { rgba: secondaryHoverColor },
      danger__resting: { rgba: dangerColor },
      danger__hover: { rgba: dangerHoverColor },
    },
  },

  shape: {
    button: { borderRadius: buttonBorderRadius },
  },
} = tokens

export const primary: ButtonToken = mergeDeepRight(button, {
  background: primaryColor,
  typography: {
    color: primaryWhite,
  },
  border: {
    type: 'border',
    style: 'solid',
    width: '1px',
    color: primaryColor,
    radius: buttonBorderRadius,
  },
  states: {
    hover: {
      background: primaryHoverColor,
    },
  },
})
export const secondary: ButtonToken = mergeDeepRight(primary, {
  background: secondaryColor,
  border: {
    color: secondaryColor,
  },
  states: {
    hover: {
      background: secondaryHoverColor,
      border: {
        color: secondaryHoverColor,
      },
    },
  },
})

export const danger: ButtonToken = mergeDeepRight(primary, {
  background: dangerColor,
  border: {
    color: dangerColor,
  },
  states: {
    hover: {
      background: dangerHoverColor,
      border: {
        color: dangerHoverColor,
      },
    },
  },
})
