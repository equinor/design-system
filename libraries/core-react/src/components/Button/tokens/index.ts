import { ButtonTokenSet } from '../Button.types'

import * as contained from './contained'
import * as outlined from './outlined'
import * as ghost from './ghost'
import * as icon from './icon'

type ButtonTokens = {
  primary: ButtonTokenSet
  secondary: ButtonTokenSet
  danger: ButtonTokenSet
}

export const token: ButtonTokens = {
  primary: {
    contained: contained.primary,
    outlined: outlined.primary,
    ghost: ghost.primary,
    ghost_icon: icon.primary,
  },
  secondary: {
    contained: contained.secondary,
    outlined: outlined.secondary,
    ghost: ghost.secondary,
    ghost_icon: icon.secondary,
  },
  danger: {
    contained: contained.danger,
    outlined: outlined.danger,
    ghost: ghost.danger,
    ghost_icon: icon.danger,
  },
}
