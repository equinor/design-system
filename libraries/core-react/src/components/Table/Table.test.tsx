/* eslint-disable no-undef */
import { render, cleanup, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'jest-styled-components'
import { Table } from '.'
import styled from 'styled-components'
import { token as dataCellToken } from './DataCell/DataCell.tokens'
import { token as headerCellToken } from './HeaderCell/HeaderCell.tokens'
import { EdsProvider } from '../EdsProvider'

const { Caption, Cell, Head, Row, Body } = Table

const trim = (x: string): string => x.replace(/ /g, '')

afterEach(cleanup)

describe('Caption', () => {
  it('Renders a table with caption element present in the document', () => {
    const { container } = render(
      <Table>
        <Caption>test</Caption>
      </Table>,
    )
    expect(container.querySelector('Caption')).toBeInTheDocument()
  })
  it('Prints the caption', () => {
    const text = "I'm da caption"
    const { getByText } = render(
      <Table>
        <Head>
          <Row>
            <Cell>{text}</Cell>
          </Row>
        </Head>
      </Table>,
    )
    expect(getByText(text)).toBeDefined()
  })
  it('Renders a table with caption, and caption-side set default to top', () => {
    const { container } = render(
      <Table>
        <Caption>test</Caption>
      </Table>,
    )
    expect(container.querySelector('Caption')).toHaveStyleRule(
      'caption-side',
      'top',
    )
  })
  it('Renders a table with caption, and caption-side set to bottom', () => {
    const { container } = render(
      <Table>
        <Caption captionSide="bottom">test</Caption>
      </Table>,
    )
    expect(container.querySelector('Caption')).toHaveStyleRule(
      'caption-side',
      'bottom',
    )
  })
})

describe('Table', () => {
  it('Can render a cell as a header cell', () => {
    const text = 'Name'
    const { getByText, container } = render(
      <Table>
        <Head>
          <Row>
            <Cell>{text}</Cell>
          </Row>
        </Head>
      </Table>,
    )

    const headerCell = getByText(text).parentElement
    const th = container.querySelector('th')
    expect(headerCell).toEqual(th)
  })

  it('Renders the header row visually different than the other rows by using a background colour', () => {
    const header = 'Header'
    const body = 'Body content'
    const { getByText } = render(
      <Table>
        <Head>
          <Row>
            <Cell>{header}</Cell>
          </Row>
        </Head>
        <Body>
          <Row>
            <Cell>{body}</Cell>
          </Row>
        </Body>
      </Table>,
    )
    const headElement = getByText(header).closest('thead')
    const regularContent = getByText(body)
    const cellBackground = trim(headerCellToken.background)
    const cellBorderBottom =
      headerCellToken.border.type === 'bordergroup'
        ? `${headerCellToken.border.bottom.width} ${
            headerCellToken.border.bottom.style
          } ${trim(headerCellToken.border.bottom.color)}`
        : ''

    expect(headElement).toHaveStyleRule('background', cellBackground)
    expect(headElement).toHaveStyleRule('border-bottom', cellBorderBottom)
    expect(regularContent).not.toHaveStyleRule('background', cellBackground)
  })

  it('Adjusts font for better readability if the text is a number', () => {
    const text = '369470'
    const { getByText } = render(
      <Table>
        <Body>
          <Row>
            <Cell variant="numeric">{text}</Cell>
          </Row>
        </Body>
      </Table>,
    )

    const trimmedFontFeature = dataCellToken.variants.numeric.typography.fontFeature.replace(
      /\s*,\s*/g,
      ',',
    )
    expect(getByText(text)).toHaveStyleRule(
      'font-feature-settings',
      trimmedFontFeature,
    )
  })

  const StyledTable = styled(Table)`
    clip-path: unset;
  `

  it('Can extend the css for the component', () => {
    const { container } = render(<StyledTable />)
    expect(container.firstChild).toHaveStyleRule('clip-path', 'unset')
  })

  it('Has correct color on active row', () => {
    render(
      <Table>
        <Body>
          <Row active data-testid="row">
            <Cell>active</Cell>
          </Row>
        </Body>
      </Table>,
    )

    const row = screen.getByTestId('row')
    expect(row).toHaveStyleRule(
      'background',
      trim(dataCellToken.states.active.background),
    )
  })
  it('Has correct color on error cell even when active', () => {
    render(
      <Table>
        <Body>
          <Row active>
            <Cell color="error">error</Cell>
          </Row>
        </Body>
      </Table>,
    )

    const cell = screen.getByText('error')
    expect(cell).toHaveStyleRule(
      'background',
      trim(dataCellToken.validation.error.background),
    )
  })
  it('Has comfortable as default density', () => {
    const cellText = 'Header content'
    const headerText = 'Cell content'
    const { getByText } = render(
      <Table>
        <Head>
          <Row>
            <Cell>{headerText}</Cell>
          </Row>
        </Head>
        <Body>
          <Row>
            <Cell>{cellText}</Cell>
          </Row>
        </Body>
      </Table>,
    )

    expect(getByText(headerText).parentElement).toHaveStyleRule(
      'height',
      headerCellToken.height,
    )
    expect(getByText(cellText)).toHaveStyleRule('height', dataCellToken.height)
  })
  it('Has compact density', () => {
    const cellText = 'Header content'
    const headerText = 'Cell content'
    const { getByText } = render(
      <EdsProvider density="compact">
        <Table>
          <Head>
            <Row>
              <Cell>{headerText}</Cell>
            </Row>
          </Head>
          <Body>
            <Row>
              <Cell>{cellText}</Cell>
            </Row>
          </Body>
        </Table>
      </EdsProvider>,
    )

    expect(getByText(headerText).parentElement).toHaveStyleRule(
      'height',
      headerCellToken.modes.compact.height,
    )
    expect(getByText(cellText)).toHaveStyleRule(
      'height',
      dataCellToken.modes.compact.height,
    )
  })
  it('Has aria-sort when sort is provided', () => {
    const headerText = 'Cell content'
    const { getByText } = render(
      <Table>
        <Head>
          <Row>
            <Cell sort="ascending">{headerText}</Cell>
          </Row>
        </Head>
      </Table>,
    )
    const headerCell = getByText(headerText).parentElement
    expect(headerCell).toHaveAttribute('aria-sort', 'ascending')
  })
  it('Has active styling when sort is ascending or descending', () => {
    const headerText1 = 'Cell content'
    const headerText2 = 'Cell content 2'
    const { getByText } = render(
      <Table>
        <Head>
          <Row>
            <Cell sort="ascending">{headerText1}</Cell>
            <Cell sort="descending">{headerText2}</Cell>
          </Row>
        </Head>
      </Table>,
    )

    const borderBottomColor = trim(
      headerCellToken.states.active.border.type === 'bordergroup'
        ? headerCellToken.states.active.border.bottom.color
        : '',
    )
    const headerCell1 = getByText(headerText1).parentElement
    const headerCell2 = getByText(headerText2).parentElement

    expect(headerCell1).toHaveAttribute('aria-sort', 'ascending')
    expect(headerCell2).toHaveAttribute('aria-sort', 'descending')

    expect(headerCell1).toHaveStyleRule(
      'background',
      trim(headerCellToken.states.active.background),
    )
    expect(headerCell1).toHaveStyleRule(
      'color',
      trim(headerCellToken.states.active.typography.color),
    )
    expect(headerCell1).toHaveStyleRule('border-color', borderBottomColor)

    expect(headerCell2).toHaveStyleRule(
      'background',
      trim(headerCellToken.states.active.background),
    )
    expect(headerCell2).toHaveStyleRule(
      'color',
      trim(headerCellToken.states.active.typography.color),
    )
    expect(headerCell2).toHaveStyleRule('border-color', borderBottomColor)
  })
})
