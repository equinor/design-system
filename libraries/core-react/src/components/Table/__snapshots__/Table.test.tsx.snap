// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Matches snapshot 1`] = `[Function]`;
